src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/NuGet/Relations/PackageToDiagnosticRelation.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections;
+
+namespace Microsoft.VisualStudio.NuGet
+{
+    [Export(typeof(IRelation))]
+    internal sealed class PackageToDiagnosticRelation : RelationBase<PackageReferenceItem, DiagnosticItem>
+    {
+        protected override bool HasContainedItems(PackageReferenceItem parent)
+        {
+            return parent.Target.Logs.Any(log => log.LibraryId == parent.Library.Name);
+        }
+
+        protected override void UpdateContainsCollection(PackageReferenceItem parent, AggregateContainsRelationCollectionSpan span)
+        {
+            span.UpdateContainsItems(
+                parent.Target.Logs.Where(log => log.LibraryId == parent.Library.Name).OrderBy(log => log.LibraryId).ThenBy(log => log.Message),
```

Comment:
This is ordered by Culture by default, is that deliberate?

Reply:
So long as it's stable, it's fine. There order here is just used for merging data structures. The UI applies its own sorting later.